In [1]:
import pandas as pd 
from Data import *
import numpy as np

In [5]:
class Indexer_performance():
    def __init__(self,indexer,queryCorpus,query_res_file:str) -> None:
        self.indexer=indexer
        self.queryCorpus=queryCorpus
        self.query_res_file=query_res_file
        self.Gt_mat=self.get_query_Gt(self.query_res_file)
    def get_query_Gt(self,query_res_file):
        file=open(query_res_file,'r')
        res=np.zeros((225,1400))
        for line in file.readlines():
            line=line.replace(' ',' ').split(' ')
            res[int(line[0])-1,int(line[1])-1]=1
        return res

    def get_query_results(self,queryCorpus,thresh,method):
        res=np.zeros((225,1400))
        for i,query in enumerate(queryCorpus):
            res[i,self.indexer.get_documents_for_query(query['text'],thresh,method)]=1
        return res

    def get_precision_recall(self,thresh,method='jaccard'):
        doc=self.get_query_results(self.queryCorpus,thresh,method)
        recall=sum(np.sum(doc*self.Gt_mat,axis=1)/np.sum(self.Gt_mat,axis=1))*100/225
        precision=sum(np.sum(doc*self.Gt_mat,axis=1)/np.sum(doc,axis=1))*100/225
        return precision,recall

In [6]:
indexer=Indexer(file_name='C:/Users/pc/Documents/programming/RI_TP/Implementation/indexer01.pkl')
queryCorpus=CranQueryCorpus("C:/Users/pc/Documents/programming/RI_TP/Datasets/Cran/cran.qry")

In [10]:
#indexer.save('indexer01.pkl')

In [7]:
id_performance=Indexer_performance(indexer,queryCorpus,'C:/Users/pc/Documents/programming/RI_TP/Datasets/Cran/cranqrel')

In [17]:
id_performance.get_precision_recall(0.132,method='Dice')

(20.26769832490872, 58.266903341252636)

Stemming with no lemmatization :(20.215267394365846, 58.266903341252636)

lemmatization no stemming :(14.004963636767473, 64.10063417182562)

Stemming and lemmatization :(19.929313146423933, 59.04080104848369)

In [236]:
doc=id_performance.get_query_results(id_performance.queryCorpus)

In [229]:
queryCorpus.get_raw_docs()[224]

'.I 365\n.W\nwhat design factors can be used to control lift-drag ratios at mach\nnumbers above 5 . \n'

In [170]:
np.where(id_performance.Gt_mat[0]==1)

(array([ 11,  12,  13,  14,  28,  29,  30,  36,  50,  51,  55,  56,  65,
         94, 101, 141, 183, 184, 194, 377, 461, 485, 496, 857, 858, 874,
        875, 878, 879], dtype=int64),)

In [193]:
doc[224].max()

0.0

In [237]:
np.sum(doc,axis=1)==0

array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False,

In [238]:
sum(np.sum(doc*id_performance.Gt_mat,axis=1)/np.sum(id_performance.Gt_mat,axis=1))*100/225

70.09893620319477

In [239]:
sum(np.sum(doc*id_performance.Gt_mat,axis=1)/np.sum(doc,axis=1))*100/225

12.237244148954323